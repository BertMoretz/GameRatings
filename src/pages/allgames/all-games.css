a.coverlink {
    background-color: rgb(0, 0, 0);
    display: inline-block;

}

a.coverlink:hover .elem{
  opacity:    0.2;
}

a.coverlink .elem{
  vertical-align:     middle;
  transition:         opacity 0.3s;
  -webkit-transition: opacity 0.3s;
}

.detailscover {
  position: absolute;
  width: 250px;
  height: 350px;
  color: rgb(250, 250, 250);
  opacity: 0;
  top: 40%;
  font-size: 28px;
  font-family: 'Roboto';
  text-align: center;
}

a.coverlink:hover .detailscover{
  opacity: 1;
}

a.coverlink:hover .caption{
  opacity: 0;
  transition:         opacity 0.3s;
  -webkit-transition: opacity 0.3s;
}

.elem {
  width: 250px;
  height: 350px;
}

.score {
    color: white;
    right: 10%
}

.gridList {
    justify-content: center;
}

.loading {
  text-align: center;
  margin-top: 24px;
}

.paging {
  text-align: center;
}

.paging ul{
  display: inline-block;
  padding-left: 15px;
  padding-right: 15px;
  font-family: 'Roboto';
  background-color: #f5f5f5;
}

.paging li {
  display: inline-block;
  padding: 15px;
}

.active {
  background-color: #e0e0e0;
}

.page {
  padding: 15px;
  cursor: pointer;
}
