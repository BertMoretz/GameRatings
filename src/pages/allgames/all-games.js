import React from 'react'
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import GridList from '@material-ui/core/GridList';
import GridListTile from '@material-ui/core/GridListTile';
import ListSubheader from '@material-ui/core/ListSubheader';

import ReactPaginate from 'react-paginate';

import Paper from '@material-ui/core/Paper';
import axios from 'axios';
import CircularProgress from '@material-ui/core/CircularProgress';

import { GameCard } from "./game-card"
import style from "./all-games.css"

export class AllGames extends React.Component {

    state = {
      games: void 0,
      offset: 0
    }

    loadGames = () => {
      axios
          .get(`https://cors-anywhere.herokuapp.com/http://api-v3.igdb.com/games/?fields=name,cover.*,genres.*,platforms,popularity,summary,aggregated_rating&limit=15&offset=${this.state.offset}&&expand=cover,genres&order=popularity:desc`, {headers: {
            "user-key": "861c079a35348acf2360c08a2efc2e90"
          }})
          .then(response => {
            console.log('Axios returned', response.data)
            this.setState({
              games: response.data
            })
          });
    }

    componentDidMount() {
      this.loadGames();
    }



    buildDetailsClickHandler = (game) => () => {
       this.props.history.push(`/game/${game.id}`)
     }

     handlePageClick = data => {
       let selected = data.selected

       let off = selected * 15;

       this.setState({ offset: off }, () => {
         this.loadGames();
       })
       console.log(this.state.offset + " " + off);
     }


    render() {

        const { games } = this.state

        if(!games) {
          return <div className={style.loading}> <CircularProgress /> </div>
        }

        return (
          <div>
            <GridList cellHeight={180} className={style.gridList} cellSpacing>
              <GridListTile key="Subheader" cols={2} style={{ height: 'auto' }}>
                <ListSubheader component="div">All Games</ListSubheader>
              </GridListTile>

              {this.state.games.map(game =>
                <GameCard
                  key={game.name}
                  game={game}
                  handleDetailsClick={this.buildDetailsClickHandler(game)}
                  />
              )}

            </GridList>
            <div className={style.paging}>
              <ReactPaginate
                previousLabel={'previous'}
                pageRangeDisplayed = {10}
                nextLabel={'next'}
                onPageChange={this.handlePageClick}
                containerClassName={'pagination'}
                subContainerClassName={'pages pagination'}
                activeClassName={style.active}
                pageLinkClassName={style.page}
              />
            </div>
          </div>
        )
    }
}
