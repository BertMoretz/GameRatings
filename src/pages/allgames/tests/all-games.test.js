import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from 'redux-mock-store';
import { MemoryRouter, BrowserRouter as Router, Route } from 'react-router-dom';

import axios from 'axios';
import moxios from 'moxios';

import {AllGames} from '../all-games'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import { shallow, mount, render } from 'enzyme';

describe('AllGames', () => {
  it('test render empty component', () => {
    const initialState = {
      games: void 0,
      offset: 0
    }

    const container = renderer
      .create(
        <Router>
          <Route component={AllGames}/>
        </Router>,
      )

    expect(container.toJSON()).toMatchSnapshot()
  }),

  test('Grid renders as container', () => {
      const props = {
        match: {
          page: 6
        }
      }

      const game = [{
        "id": 36950,
        "aggregated_rating": 61.0,
        "cover": {
          "id": 71030,
          "alpha_channel": false,
          "animated": false,
          "game": 36950,
          "height": 960,
          "image_id": "co1it2",
          "url": "//images.igdb.com/igdb/image/upload/t_thumb/co1it2.jpg",
          "width": 677
        },
        "genres": [
          {
            "id": 5,
            "created_at": 1297555200,
            "name": "Shooter",
            "slug": "shooter",
            "updated_at": 1323216000,
            "url": "https://www.igdb.com/genres/shooter"
          },
          {
            "id": 12,
            "created_at": 1297555200,
            "name": "Role-playing (RPG)",
            "slug": "role-playing-rpg",
            "updated_at": 1323216000,
            "url": "https://www.igdb.com/genres/role-playing-rpg"
          },
          {
            "id": 31,
            "created_at": 1323561600,
            "name": "Adventure",
            "slug": "adventure",
            "updated_at": 1323561600,
            "url": "https://www.igdb.com/genres/adventure"
          }
        ],
        "name": "Anthem",
        "platforms": [
          6,
          48,
          49
        ],
        "summary": "Anthem is a shared-world action RPG, where players can delve into a vast landscape teeming with amazing technology and forgotten treasures. This is a world where Freelancers are called upon to defeat savage beasts, ruthless marauders, and forces plotting to conquer humanity."
      }]

      const shallowComponent = shallow(<AllGames {...props}/>)
      shallowComponent.setState({games:  game})
      expect(shallowComponent).toBeDefined()
      expect(shallowComponent.debug()).toMatchSnapshot()

  	    // Test lifecycle method calls and all possible component states
  	    // console.log(shallowComponent.debug())
  	    // console.log(mountComponent.debug())
  }),

  it('test api', () => {

      let onFulfilled = jest.fn()

      moxios.stubRequest('https://cors-anywhere.herokuapp.com/http://api-v3.igdb.com/games', {
          status: 200,
          response: {
              results: [{ test: 'game' }]
          }
      })

      axios.get('https://cors-anywhere.herokuapp.com/http://api-v3.igdb.com/games').then(onFulfilled)

      moxios.wait(function () {
        expect(onFulfilled).toBeCalled();
        expect(onFulfilled.mock.calls[0][0]).toMatchSnapshot();
        done()
      })
  })

  afterEach(() => {
      moxios.uninstall()
  })
})
