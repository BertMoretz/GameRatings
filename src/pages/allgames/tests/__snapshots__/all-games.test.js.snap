// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllGames Grid renders as container 1`] = `
"<div>
  <WithStyles(GridList) cellHeight={180} className=\\"gridList\\" cellSpacing={true}>
    <WithStyles(GridListTile) cols={2} style={{...}}>
      <WithStyles(ListSubheader) component=\\"div\\">
        All Games
      </WithStyles(ListSubheader)>
    </WithStyles(GridListTile)>
    <GameCard game={{...}} handleDetailsClick={[Function]} />
  </WithStyles(GridList)>
  <div className=\\"paging\\">
    <PaginationBoxView previousLabel=\\"previous\\" pageRangeDisplayed={10} nextLabel=\\"next\\" onPageChange={[Function]} containerClassName=\\"pagination\\" subContainerClassName=\\"pages pagination\\" activeClassName=\\"active\\" pageLinkClassName=\\"page\\" pageCount={10} marginPagesDisplayed={3} previousClassName=\\"previous\\" nextClassName=\\"next\\" breakLabel=\\"...\\" disabledClassName=\\"disabled\\" disableInitialCallback={false} />
  </div>
</div>"
`;

exports[`AllGames test render empty component 1`] = `
<div
  className="loading"
>
   
  <div
    className="MuiCircularProgress-root-1 MuiCircularProgress-colorPrimary-4 MuiCircularProgress-indeterminate-3"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-6"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-7 MuiCircularProgress-circleIndeterminate-9"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
   
</div>
`;
